pkg/sql/schemachanger/scplan/internal/rules/dep_add_column.go

```diff
@@ -108,3 +108,47 @@ func init() {
 			}
 		})
 }
+
+// These rules ensure secondary indexes added as a result of adding columns
+// progress into public in accordance with the new primary index.
+func init() {
+	// The new primary index should reach VALIDATED, a ready state to be used as
+	// source to backfill the new secondary index, before the secondary index becomes existent.
+	registerDepRule(
+		"primary index with new columns should exist before secondary indexes",
```

Comment:
This is an oversight on my part when I moved these rule definitions about, I think, but I just noticed that "new columns" should not be mentioned at all in this rule name. This rule can come into play regardless of columns. Instead, it might be worth mentioning why this rule is required for ADD COLUMN UNIQUE in the commentary above.